src/test/java/com/google/firebase/auth/FirebaseAuthIT.java

```diff
@@ -140,37 +141,98 @@ public void testDeleteNonExistingUser() throws Exception {
+      assertEquals("Random User", user.getDisplayName());
+      assertEquals(randomUser.email, user.getEmail());
+      assertEquals(randomPhoneNumber, user.getPhoneNumber());
+      assertEquals("https://example.com/photo.png", user.getPhotoUrl());
+      assertTrue(user.isEmailVerified());
+      assertFalse(user.isDisabled());
+
+      assertEquals(2, user.getProviderData().length);
       List<String> providers = new ArrayList<>();
-      for (UserInfo provider : userRecord.getProviderData()) {
+      for (UserInfo provider : user.getProviderData()) {
         providers.add(provider.getProviderId());
       }
       assertTrue(providers.contains("password"));
       assertTrue(providers.contains("phone"));
 
-      checkRecreate(randomUser.uid);
+      checkRecreate(user.getUid());
     } finally {
-      auth.deleteUserAsync(userRecord.getUid()).get();
+      auth.deleteUserAsync(randomUser.uid).get();
+    }
+  }
+
+  @Test
+  public void testLookupUserByPhone() throws Exception {
+    RandomUser randomUser1 = RandomUser.create();
+    String phoneNumber1 = null;
+    RandomUser randomUser2 = RandomUser.create();
+    String randomPhoneNumber2 = randomPhoneNumber();
+    RandomUser randomUser3 = RandomUser.create();
+    String randomPhoneNumber3 = randomPhoneNumber();
+    try {
+      UserRecord user1 = createUser(
+          randomUser1.uid, /* phoneNumber= */ null, randomUser1.email);
+      UserRecord user2 = createUser(
+          randomUser2.uid, randomPhoneNumber2, randomUser2.email);
+      UserImportResult user3 = importUser(
+          randomUser3.uid, randomPhoneNumber3, randomUser3.email, "google.com");
+
+      UserRecord lookedUpRecord = auth.getUserByPhoneNumberAsync(randomPhoneNumber2).get();
+      assertEquals(lookedUpRecord.getUid(), randomUser2.uid);
+
+      lookedUpRecord = auth.getUserByPhoneNumberAsync(randomPhoneNumber3).get();
+      assertEquals(lookedUpRecord.getUid(), randomUser3.uid);
+    } finally {
+      auth.deleteUserAsync(randomUser1.uid).get();
+      auth.deleteUserAsync(randomUser2.uid).get();
+      auth.deleteUserAsync(randomUser3.uid).get();
+    }
+  }
+
+  @Test
+  public void testLookupUserByFederatedId() throws Exception {
+    RandomUser randomUser1 = RandomUser.create();
+    String phoneNumber1 = null;
+    RandomUser randomUser2 = RandomUser.create();
+    String randomPhoneNumber2 = randomPhoneNumber();
+    RandomUser randomUser3 = RandomUser.create();
+    String randomPhoneNumber3 = randomPhoneNumber();
+    try {
+      UserRecord user1 = createUser(
+          randomUser1.uid, /* phoneNumber= */ null, randomUser1.email);
+      UserRecord user2 = createUser(
+          randomUser2.uid, randomPhoneNumber2, randomUser2.email);
+      UserImportResult user3 = importUser(
+          randomUser3.uid, randomPhoneNumber3, randomUser3.email, "google.com");
+
+      UserRecord lookedUpRecord = auth.getUserByFederatedIdAsync(
+          randomUser3.uid + "_google.com", "google.com").get();
```

Comment:
It's not clear why `randomUser3.uid + "_google.com"` is correct. Maybe add that as an explicit paramter to the importUser method? i.e. (on line 210-211):

```java
UserImportResult user3 = importUser(
    randomUser3.uid, randomPhoneNumber3, randomUser3.email,
    "google.com", randomUser3.uid + "_google.com");
```